Build failure: botan3 (aarch64-darwin)

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *botan3* on aarch64-darwin

### Build log

```console
$ nix build 'nixpkgs#botan3'
error:
       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'botan-3.6.1'
         whose name attribute is located at /nix/store/np74wqp8a8jzss4z38ch34mmivl6id9k-source/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'buildInputs' of derivation 'botan-3.6.1'
         at /nix/store/np74wqp8a8jzss4z38ch34mmivl6id9k-source/pkgs/stdenv/generic/make-derivation.nix:383:7:
          382|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          383|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          384|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: Package ‘jitterentropy-3.6.0’ in /nix/store/np74wqp8a8jzss4z38ch34mmivl6id9k-source/pkgs/development/libraries/jitterentropy/default.nix:27 is not available on the requested hostPlatform:
         hostPlatform.config = "aarch64-apple-darwin"
         package.meta.platforms = [
           "aarch64-linux"
           "armv5tel-linux"
           "armv6l-linux"
           "armv7a-linux"
           "armv7l-linux"
           "i686-linux"
           "loongarch64-linux"
           "m68k-linux"
           "microblaze-linux"
           "microblazeel-linux"
           "mips-linux"
           "mips64-linux"
           "mips64el-linux"
           "mipsel-linux"
           "powerpc64-linux"
           "powerpc64le-linux"
           "riscv32-linux"
           "riscv64-linux"
           "s390-linux"
           "s390x-linux"
           "x86_64-linux"
         ]
         package.meta.badPlatforms = [ ]
       , refusing to evaluate.
```

With `NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1`:

```console
$ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1
$ nix build 'nixpkgs#botan3' --impure
error: builder for '/nix/store/lx9gxlr7jh4nd7p5i1nx0l3d8jyqmmyn-jitterentropy-3.6.0.drv' failed with exit code 2;
       last 25 log lines:
       >         ^
       > 1 warning generated.
       > In file included from src/jitterentropy-base.c:33:
       > In file included from src/jitterentropy-gcd.h:23:
       > In file included from src/jitterentropy-internal.h:45:
       > In file included from ./jitterentropy.h:148:
       > ./jitterentropy-base-user.h:125:9: warning: extension used [-Wlanguage-extension-token]
       >         asm volatile("mrs %0, cntvct_el0" : "=r" (ctr_val));
       >         ^
       > 1 warning generated.
       > In file included from src/jitterentropy-timer.c:22:
       > In file included from src/jitterentropy-timer.h:23:
       > In file included from src/jitterentropy-internal.h:45:
       > In file included from ./jitterentropy.h:148:
       > ./jitterentropy-base-user.h:125:9: warning: extension used [-Wlanguage-extension-token]
       >         asm volatile("mrs %0, cntvct_el0" : "=r" (ctr_val));
       >         ^
       > 1 warning generated.
       > 1 warning generated.
       > 1 warning generated.
       > clang -shared -Wl,-soname,libjitterentropy.so.3 -o libjitterentropy.so.3.6.0 jitterentropy-base.o jitterentropy-gcd.o jitterentropy-health.o jitterentropy-noise.o jitterentropy-sha3.o jitterentropy-timer.o -Wl,-z,relro,-z,now -lpthread  -lrt
       > ar rcs libjitterentropy.a jitterentropy-base.o jitterentropy-gcd.o jitterentropy-health.o jitterentropy-noise.o jitterentropy-sha3.o jitterentropy-timer.o
       > ld: unknown option: -soname
       > clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
       > make: *** [Makefile:69: libjitterentropy.so.3.6.0] Error 1
       For full logs, run 'nix log /nix/store/lx9gxlr7jh4nd7p5i1nx0l3d8jyqmmyn-jitterentropy-3.6.0.drv'.
error: 1 dependencies of derivation '/nix/store/1k3rh1yh19wb7gpnzqhinqsrxspdwvyr-botan-3.6.1.drv' failed to build
```

### Additional context

`jitterentropy` dependency was added in https://github.com/NixOS/nixpkgs/commit/3d69d435ba19175a76d733c96ad09a2f93483e95

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@7c6f434c
@thillux 

### Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.6.0, macOS 14.7.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/np74wqp8a8jzss4z38ch34mmivl6id9k-source`

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/jitterentropy/default.nix
+++ b/pkgs/development/libraries/jitterentropy/default.nix
@@ -17,12 +17,12 @@ stdenv.mkDerivation rec {
 
   hardeningDisable = [ "fortify" ]; # avoid warnings
 
-  meta = with lib; {
+  meta = {
     description = "Provides a noise source using the CPU execution timing jitter";
     homepage = "https://github.com/smuellerDD/jitterentropy-library";
     changelog = "https://github.com/smuellerDD/jitterentropy-library/raw/v${version}/CHANGES.md";
-    license = with licenses; [ bsd3 /* OR */ gpl2Only ];
-    platforms = platforms.linux ++ platforms.darwin;
-    maintainers = with maintainers; [ johnazoidberg c0bw3b ];
+    license = with lib.licenses; [ bsd3 /* OR */ gpl2Only ];
+    platforms = lib.platforms.linux ++ lib.platforms.darwin;
+    maintainers = with lib.maintainers; [ johnazoidberg c0bw3b ];
   };
 }

--- a/pkgs/development/libraries/jitterentropy/default.nix
+++ b/pkgs/development/libraries/jitterentropy/default.nix
@@ -1,4 +1,9 @@
-{ lib, stdenv, cmake, fetchFromGitHub }:
+{
+  lib,
+  stdenv,
+  cmake,
+  fetchFromGitHub,
+}:
 
 stdenv.mkDerivation rec {
   pname = "jitterentropy";
@@ -13,7 +18,10 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake ];
 
-  outputs = [ "out" "dev" ];
+  outputs = [
+    "out"
+    "dev"
+  ];
 
   hardeningDisable = [ "fortify" ]; # avoid warnings
 
@@ -21,8 +29,14 @@ stdenv.mkDerivation rec {
     description = "Provides a noise source using the CPU execution timing jitter";
     homepage = "https://github.com/smuellerDD/jitterentropy-library";
     changelog = "https://github.com/smuellerDD/jitterentropy-library/raw/v${version}/CHANGES.md";
-    license = with lib.licenses; [ bsd3 /* OR */ gpl2Only ];
+    license = with lib.licenses; [
+      bsd3 # OR
+      gpl2Only
+    ];
     platforms = lib.platforms.linux ++ lib.platforms.darwin;
-    maintainers = with lib.maintainers; [ johnazoidberg c0bw3b ];
+    maintainers = with lib.maintainers; [
+      johnazoidberg
+      c0bw3b
+    ];
   };
 }

--- a/pkgs/development/libraries/jitterentropy/default.nix
+++ b/pkgs/development/libraries/jitterentropy/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub }:
+{ lib, stdenv, cmake, fetchFromGitHub }:
 
 stdenv.mkDerivation rec {
   pname = "jitterentropy";
@@ -11,24 +11,18 @@ stdenv.mkDerivation rec {
     hash = "sha256-CPvgc/W5Z2OfbP9Lp2tQevUQZr+xlh6q5r5Fp2WUHhg=";
   };
 
+  nativeBuildInputs = [ cmake ];
+
   outputs = [ "out" "dev" ];
 
-  enableParallelBuilding = true;
   hardeningDisable = [ "fortify" ]; # avoid warnings
-
-  # prevent jitterentropy from builtin strip to allow controlling this from the derivation's
-  # settings. Also fixes a strange issue, where this strip may fail when cross-compiling.
-  installFlags = [
-    "INSTALL_STRIP=install"
-    "PREFIX=${placeholder "out"}"
-  ];
 
   meta = with lib; {
     description = "Provides a noise source using the CPU execution timing jitter";
     homepage = "https://github.com/smuellerDD/jitterentropy-library";
     changelog = "https://github.com/smuellerDD/jitterentropy-library/raw/v${version}/CHANGES.md";
     license = with licenses; [ bsd3 /* OR */ gpl2Only ];
-    platforms = platforms.linux;
+    platforms = platforms.linux ++ platforms.darwin;
     maintainers = with maintainers; [ johnazoidberg c0bw3b ];
   };
 }

